### PR TITLE
CAMEL-14826: Add unit tests to the platform-http component

### PR DESCRIPTION
**Pull request by zbendhiba**
A test of platform-http using jetty server. For the test, I needed to create a custom Consumer.

### CODE REVIEW

File: components/camel-platform-http/src/test/java/org/apache/camel/component/platform/http/PlatformHttpTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                from("platform-http:/post")
+                        .transform().body(String.class, b -> b.toUpperCase());
+            }
+        });
+        ctx.start();
+    }
+
+    @AfterAll
+    public static void initPort() throws Exception {
+        ctx.stop();
```
Comment (aldettinger) [inline components/camel-platform-http/src/test/java/org/apache/camel/component/platform/http/PlatformHttpTest.java:66]: I would change the name of the method, e.g. `tearDown()`

File: components/camel-platform-http/src/test/resources/log4j2.properties [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+appender.file.layout.type = PatternLayout
+appender.file.layout.pattern = %d [%-15.15t] %-5p %-30.30c{1} - %m%n
+appender.out.type = Console
+appender.out.name = out
+appender.out.layout.type = PatternLayout
+appender.out.layout.pattern = [%30.30t] %-30.30c{1} %-5p %m%n
+
+rootLogger.level = INFO
+rootLogger.appenderRef.file.ref = file
+#rootLogger.appenderRef.out.ref = out
```
Comment (aldettinger) [inline components/camel-platform-http/src/test/resources/log4j2.properties:29]: We may remove this line if not needed ?

File: components/camel-platform-http/src/test/java/org/apache/camel/component/platform/http/PlatformHttpTest.java (hunk 1 above)
Comment (zbendhiba) [reply components/camel-platform-http/src/test/java/org/apache/camel/component/platform/http/PlatformHttpTest.java:66]: yes!!

File: components/camel-platform-http/src/test/resources/log4j2.properties (hunk 2 above)
Comment (zbendhiba) [reply components/camel-platform-http/src/test/resources/log4j2.properties]: done!!

File: components/camel-platform-http/src/test/java/org/apache/camel/component/platform/http/PlatformHttpTest.java (hunk 1 above)
Comment (zbendhiba) [reply components/camel-platform-http/src/test/java/org/apache/camel/component/platform/http/PlatformHttpTest.java:65]: done !!

**Pull request merged**